xmppserver/src/main/java/org/jivesoftware/openfire/keystore/OpenfireX509TrustManager.java

```diff
@@ -274,6 +277,30 @@ protected CertPath checkChainTrusted( CertSelector selector, X509Certificate...
             pathBuilder = CertPathBuilder.getInstance( "PKIX" );
         }
 
+        if (checkRevocation) {
+            // Configure revocation checking - using default OCSP preference (OCSP before CRL)
+            PKIXRevocationChecker revChecker = (PKIXRevocationChecker)pathBuilder.getRevocationChecker();
+
+            EnumSet<PKIXRevocationChecker.Option> options = EnumSet.noneOf(PKIXRevocationChecker.Option.class);
+
+            // Only verify revocation status of end-entity (leaf) certificates if configured
+            // This avoids issues with chains where the root certificate isn't included
+            // in the chain (e.g. Let's Encrypt) and its CRL distribution point isn't accessible
```

Comment:
This comment seems wrong.

The chain checked is presumably after path-building, so if the old Trust Anchor is not included in the peer chain, we'll have used our own copy anyway, surely?

(Exception being if DANE is in play, but then we possibly don't do CRL checking anyway, and Openfire doesn't yet do DANE).

Is there a better explanation of EE-only status checking?

Reply:
Thanks Dave, your comment led me to realise I hadn't understood what was going on.